File: leetcode/0053-maxSubArray.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        for i in range(1, len(nums)):
            if nums[i-1] > 0:
                nums[i] += nums[i-1]
        return max(nums)
        
def main():
    while True:
        user_input = input("Enter your input like 1, 2, 3: ")
        char_list = user_input.split(",")
        int_list = [int(char_list[i]) for i in range(len(char_list))]
        s = Solution()
        res = s.maxSubArray(int_list)
        print("res is:", res)

if __name__ == "__main__":
    main()

File: leetcode/0053-maxSubArray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if (nums == null || nums.length == 0)
            return 0;
        int res = nums[0], curMax = nums[0], len = nums.length;
        for (int i = 1; i < len; ++i) {
            curMax = curMax > 0 ? curMax + nums[i] : nums[i];
            res = Math.max(curMax, res);
        }
        return res;
    }
}
